.github/workflows/release-as.yml

```diff
@@ -0,0 +1,38 @@
+name: Release As
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Major.Minor.Patch'
+        required: true
+
+jobs:
+  release-as:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get latest tag
+        id: latest-tag
+        run: echo "tag=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT
+      - uses: jackbilestech/semver-compare@1.0.4
+        with:
+          head: ${{ inputs.version }}
+          base: ${{ steps.latest-tag.outputs.tag }}
+          operator: '>'
+      # https://github.com/googleapis/release-please?tab=readme-ov-file#how-do-i-change-the-version-number
+      - name: Release as 
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+          git switch -c release-as-v${{ github.event.inputs.version }}
+          git commit --allow-empty -m "chore: release ${{ github.event.inputs.version }}" -m "Release-As: ${{ github.event.inputs.version }}"
+          git push origin release-as-v${{ github.event.inputs.version }}
+          gh pr create --title "chore: Release as v${{ github.event.inputs.version }}" --body "Release as v${{ github.event.inputs.version }}" --base main
+          gh pr merge --squash --delete-branch
```

Comment:
Just curious, how does this work with branch protections on main? Is there an exception configured for `github-actions[bot]` specifically?

Reply:
It shouldn't contradict the branch protection since it only merges the changes as a PR. There are no exceptions allowed in the protection rules.

Reply:
The doc [here](https://cli.github.com/manual/gh_pr_merge) states:

>  If required checks have not yet passed, auto-merge will be enabled.

I think that means this command sets the PR up to automatically merge on a manual approval, correct?
Or are approval requirements distinct from required checks?

Reply:
i believe approval requirements do not fall under this category.

thank you for this call out though, there was an issue when it came to the required status checks. since github [actions can't trigger other actions](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow#triggering-a-workflow-from-a-workflow) merging the PR through this action was failing.

a new org PAT was created. updated in [f2df914](https://github.com/passageidentity/passage-flex-rust/pull/24/commits/f2df91472819b5efdc54024d879baba4ee4075b3)